src/MomentoCacheClient.php

```diff
@@ -416,7 +416,48 @@ public function exists(mixed $key, mixed ...$other_keys): Redis|int|bool
      */
     public function expire(string $key, int $timeout, ?string $mode = null): Redis|bool
     {
-        throw MomentoToPhpRedisExceptionMapper::createCommandNotImplementedException(__FUNCTION__);
+        // Mode 'nx' is not supported
+        if ($mode === 'nx') {
+            throw MomentoToPhpRedisExceptionMapper::createArgumentNotSupportedException(__FUNCTION__, "nx");
+        }
+
+        $ttl = $timeout * 1000; // convert to milliseconds
+
+        // Mode 'lt' - decrease TTL if the new TTL is less than current
+        if ($mode === 'lt') {
+            $result = $this->client->decreaseTtl($this->cacheName, $key, $ttl);
+            if ($result->asSet()) {
+                return true;
+            } elseif ($result->asNotSet()) {
+                return false;
+            } elseif ($result->asMiss()) {
+                return false;
+            } else {
+                return MomentoToPhpRedisExceptionMapper::mapExceptionElseReturnFalse($result);
+            }
```

Comment:
(Here and elsewhere) We should be checking that it's an error explicitly and then handling the default case (none of the above). Can address in a follow up PR.